--- a/checksec
+++ b/checksec
@@ -379,6 +379,8 @@
     FS_libc=/lib/x86_64-linux-gnu/libc.so.6
   elif [[ -e /lib/arm-linux-gnueabihf/libc.so.6 ]]; then
     FS_libc=/lib/arm-linux-gnueabihf/libc.so.6
+  elif [[ -e /lib/sw_64-linux-gnu/libc.so.6 ]]; then
+    FS_libc=/lib/sw_64-linux-gnu/libc.so.6
   elif [[ -e /lib/aarch64-linux-gnu/libc.so.6 ]]; then
     FS_libc=/lib/aarch64-linux-gnu/libc.so.6
   else
@@ -430,6 +432,8 @@
       FS_libc=/lib/i386-linux-gnu/libc.so.6
     elif [[ -e /lib/x86_64-linux-gnu/libc.so.6 ]]; then
       FS_libc=/lib/x86_64-linux-gnu/libc.so.6
+    elif [[ -e /lib/sw_64-linux-gnu/libc.so.6 ]]; then
+      FS_libc=/lib/sw_64-linux-gnu/libc.so.6
     else
       printf "\033[31mError: libc not found.\033[m\n\n"
       exit 1
@@ -824,6 +828,8 @@
     FS_libc=/lib/aarch64-linux-gnu/libc.so.6
   elif [[ -e /usr/x86_64-gentoo-linux-musl/bin/ld ]]; then
     FS_libc=/usr/x86_64-gentoo-linux-musl/bin/ld
+  elif [[ -e /lib/sw_64-linux-gnu/libc.so.6 ]]; then
+    FS_libc=/lib/sw_64-linux-gnu/libc.so.6
   elif [[ -e /usr/lib/loongarch64-linux-gnu/libc.so.6 ]]; then
     FS_libc=/usr/lib/loongarch64-linux-gnu/libc.so.6
   else
